sdk/tests/core/CMakeLists.txt

```diff
@@ -3,11 +3,111 @@
 
 cmake_minimum_required (VERSION 3.10)
 
-project (az_core_test LANGUAGES C)
+project (az_core_test_lib_test LANGUAGES C)
```

Comment:
Could you tell how this will look like once the files for mqtt5 are moved out of core? For use to understand if this construct will be permanent and how it would like?

Reply:
If we decide to name the move folder `mqtt` there would be the following targets:
- `az_mqtt`: The project the customer would consume
- `az_mqtt_test`: The project the tests would consume